source/client-side-encryption/client-side-encryption.rst

```diff
@@ -649,33 +649,44 @@ FLE 2 ``CreateCollection()`` and ``Collection.Drop()``
-  If ``EncryptedFieldConfig["escCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.esc``.
-  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldConfigMap``.
-- Create the collection with name ``EncryptedFieldConfig["eccCollection"]`` using default options.
-  If ``EncryptedFieldConfig["eccCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.ecc``.
-  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldConfigMap``.
-- Create the collection with name ``EncryptedFieldConfig["ecocCollection"]`` using default options.
-  If ``EncryptedFieldConfig["ecocCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.ecoc``.
-  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldConfigMap``.
-- Create the collection ``collectionName`` with ``collectionOptions`` and the option ``encryptedFields`` set to the ``EncryptedFieldConfig``.
+Drivers MUST have a BSON document option named ``EncryptedFields`` in ``CreateCollection()``.
+
+A call to a driver helper ``CreateCollection(collectionName, collectionOptions)`` must check if the collection namespace (``<databaseName>.<collectionName>``) has an associated ``EncryptedFields``. Check for an associated ``EncryptedFields`` from the following:
+
+- The ``EncryptedFields`` option passed in ``collectionOptions``.
+- The value of ``AutoEncryptionOpts.EncryptedFieldsMap[<databaseName>.<collectionName>]``.
+
+If the collection namespace has an associated ``EncryptedFields``, then do the following operations. If any of the following operations error, the remaining operations are not attempted:
+
+- Create the collection with name ``EncryptedFields["escCollection"]`` using default options.
+  If ``EncryptedFields["escCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.esc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldsMap``.
+- Create the collection with name ``EncryptedFields["eccCollection"]`` using default options.
+  If ``EncryptedFields["eccCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.ecc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldsMap``.
+- Create the collection with name ``EncryptedFields["ecocCollection"]`` using default options.
+  If ``EncryptedFields["ecocCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.ecoc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldsMap``.
+- Create the collection ``collectionName`` with ``collectionOptions`` and the option ``encryptedFields`` set to the ``EncryptedFields``.
```

Comment:
Do drivers need to do anything special if the `collectionName` collection already exists?

Reply:
I think not. The server will error:

```
> db.runCommand({create: "foo"})
{
        "ok" : 0,
        "errmsg" : "Collection test.foo already exists.",
        "code" : 48,
        "codeName" : "NamespaceExists",
        "$clusterTime" : {
                "clusterTime" : Timestamp(1650500294, 1),
                "signature" : {
                        "hash" : BinData(0,"AAAAAAAAAAAAAAAAAAAAAAAAAAA="),
                        "keyId" : NumberLong(0)
                }
        },
        "operationTime" : Timestamp(1650500294, 1)
}
```